ci/infra/aws/worker-instance.tf

```diff
@@ -6,12 +6,12 @@ resource "aws_instance" "nodes" {
   key_name                    = aws_key_pair.kube.key_name
   source_dest_check           = false
   user_data                   = data.template_cloudinit_config.cfg.rendered
-  iam_instance_profile        = length(var.iam_profile_worker) == 0 ? local.aws_iam_policy_worker_terraform : var.iam_profile_worker
+  iam_instance_profile        = length(var.iam_profile_worker) == 0 ? local.aws_iam_role_policy_worker_terraform : var.iam_profile_worker
   subnet_id                   = aws_subnet.private.id
 
   depends_on = [
     aws_route.private_nat_gateway,
-    aws_iam_policy.worker,
+    aws_iam_role_policy.worker,
```

Comment:
aws_iam_instance_profile.worker instead?

Reply:
changed and tested.